packages/manager/src/components/RegionSelect/RegionSelect.types.ts

```diff
@@ -40,7 +54,7 @@ export interface RegionSelectProps<
   disabledRegions?: Record<string, DisableRegionOption>;
   helperText?: string;
   label?: string;
-  regionFilter?: RegionSite;
+  regionFilter?: RegionFilterValue;
```

Comment:
Just a thought (not blocking) 💭 

I know this would change a lot, but I wonder if `RegionFilterValue` is the best option. 

I wonder if we could use filter functions and share them throughout the app

```suggestion
  regionFilter?: (region: Region) => boolean;
```

For example:
```
export const distributedRegionsFilter = (region: Region) => region.site_type === 'distributed';
```

Not sure if this approach would be better or worse

Reply:
Good idea, maybe this can be explored as part of M3-8292